### PR TITLE
fix(gatsby-plugin-contentful): wait for contentTypes to be created)

### PR DESCRIPTION
**Pull request by wardpeet**
### Documentation
I've been investigating a large contentful site with 930k nodes. It seems to use much memory in the onCreateNode api. The site is using gatsby-transformer-remark to convert content into markdown. We don't queue/batch createNode API calls, so in the end, we run 930k nodes through remark, which takes up a lot of memory.

This approach limits the createNodeApi per contentType and asset to reduce the number of nodes in flight. Limiting this API lowers the memory from around 15gb to a steady 4gb. I haven't measured the difference in timing yet. I might even think this is faster for more significant sites as the CPU doesn't have to switch operations that often.

### CODE REVIEW

File: packages/gatsby-source-contentful/src/gatsby-node.js [hunk 1]
```diff
@@ -207,35 +207,40 @@ exports.sourceNodes = async (
...
-      locales,
-      space,
-      useNameForId: pluginConfig.get(`useNameForId`),
-      richTextOptions: pluginConfig.get(`richText`),
-    })
-  })
+  for (let i = 0; i < contentTypeItems.length; i++) {
+    const contentTypeItem = contentTypeItems[i]
+    await Promise.all(
+      normalize.createNodesForContentType({
```
Comment (vladar) [inline packages/gatsby-source-contentful/src/gatsby-node.js]: I think we need a comment here explaining _why_ we do it this way.

**Comment by KyleAMathews** [comment]
Any reason not to do this in core vs. the plugin?

**Comment by KyleAMathews** [comment]
(great find!)

**Comment by wardpeet** [comment]
We could move this to core, that means adding a createNode Queue and batching it up there. I stumbled upon this during a large contentful site and I haven't measured possible "speed" implications yet. So that's the reason why I fixed it here and not in core as I don't know what other side effects batching could bring.

Good question

**Comment by KyleAMathews** [comment]
Cool cool — I'm totally fine merging this & then looking into moving this into core at another time. With benchmark sites up, it'll be a lot easier to both see whether other source/transform stacks run into this + the speed implications.

**Pull request merged**